xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentfield.vm

```diff
@@ -22,63 +22,47 @@
-#end
-
-## Forces the value of the comment to the content from the request if it exists.
+## If a content is provided, we use it. If a comment number is provided, we use the content of the corresponding 
+## comment. Otherwise, the content is an empty string.
 #if ($request.content)
+  ## Forces the value of the comment to the content from the request if it exists (for instance in case of error 
+  ## during the captcha validation).
   #set ($value = $request.content)
+#elseif ($number && $number > -1)
+  #set ($value = $doc.getObject($xCommentClass, $number).getValue('comment'))
+#else
+  #set ($value = '')
 #end
 
-## We can't simply call $doc.display() on this object because we would get comment as the
-## field name. Instead we want the field name to be XWiki.XWikiComments_comment for new comments
-## and XWiki.XWikiComments_N_comment for edited comments.
-## To achieve this we add a new property to the comment class by copying the comment property and changing the name.
-#set ($commentObjClass = $commentObj.getxWikiClass())
-#set ($commentObjBaseClass = $commentObjClass.getXWikiClass())
-#set ($commentField = $commentClass.getXWikiClass().get('comment'))
-
-## In theory this should never occur, but apparently it happened in XWIKI-18717 in weird conditions,
-## so we're bullet-proofing
-#if ("$!commentField" != '')
-#set ($discard = $commentObjBaseClass.put($name, $commentField))
-#set ($discard = $commentObjBaseClass.get($name).setName($name))
-
-## attribution of the value to the new field in the object.
-#set ($discard = $commentObj.set($name, $value))
-
-#set ($fieldToDisplay = $commentObjClass.get($name))
-
 ## We need to know the type of editor used for comments in order to hide/display the preview button.
 #set ($defaultEditorId = $services.edit.syntaxContent.defaultEditorId)
 <input type='hidden' name='defaultEditorId' value="$escapetool.xml($defaultEditorId)" />
 
-<label for="$name">$services.localization.render('core.viewers.comments.add.comment.label')</label>
+#if ($number && $number > -1)
+  #set ($commentPrefix = "XWiki.XWikiComments_${number}")
+#else
+  #set ($commentPrefix = 'XWiki.XWikiComments')
+#end
+<label for="${commentPrefix}_comment">$services.localization.render('core.viewers.comments.add.comment.label')</label>
 #initRequiredSkinExtensions()
-## display of the comment field
-$doc.displayEdit($fieldToDisplay, '', $commentObj)
-
-## ensure to load the appropriate JSX for the editor.
+## Display of the comment field.
+#set ($commentProperty = $commentClass.get('comment'))
+#set ($properties = {
+  'id': "${commentPrefix}_comment",
+  'name': "${commentPrefix}_comment",
+  'rows': $commentProperty.getProperty('rows').value,
+  'cols': $commentProperty.getProperty('cols').value
+})
+#if ($commentProperty.getPropertyClass().isWysiwyg($xcontext.context))
+  $services.edit.syntaxContent.wysiwyg($value, $doc.syntax, $properties)
+#else
+  $services.edit.syntaxContent.text($value, $doc.syntax, $properties)
+#end
```

Comment:
Opening a PR because I'm wondering if this can be simplified further.

Reply:
I don't think this is a good idea. Any xproperty, including ``comment``, can have a **custom displayer**:

* either coming from the property type (e.g. ``TextArea``), in which case it can be defined in Java, e.g. ``TextAreaClass``, or in a Velocity template such as ``displayer_date.vm`` for Date properties
* or coming from the xclass, e.g. ``XWiki.XWikiComments``, in which case it is defined using the ``customDisplay`` meta property

What you propose is basically to bypass the custom displayer for the ``comment`` property. I don't see the motivation behind this. In any case, such a change deserves a discussion (involving also developers that do heavy customizations on top of XWiki and which rely often on custom displayers).

Reply:
I, too, am really not a fan of bypassing part of the property display process.

Reply:
Indeed I forgot the custom displayers. Thanks for reminding me of them.
In this case I propose to define somewhere (to be determined) a method allowing to use the displayer of a field, while being able to override its value and name.

Reply:
> while being able to override its value

And being very careful which author is used to execute this displayer ;)

Reply:
I've updated the field rendering logic [here](https://github.com/xwiki/xwiki-platform/pull/1757/files#diff-4883aafa61b3e034573bc2799b4cf5f9352db3f540c3c9516282d0aa12c6d558R56-R60). A new 'in-memory' object is created and initialized with the value submitted by the user.